core/src/com/etheller/warsmash/viewer5/handlers/w3x/simulation/abilities/skills/orc/chieftain/CAbilityReincarnation.java

```diff
@@ -0,0 +1,92 @@
+package com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.skills.orc.chieftain;
+
+import com.etheller.warsmash.units.manager.MutableObjectData.MutableGameObject;
+import com.etheller.warsmash.util.War3ID;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.CSimulation;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.CUnit;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.generic.AbstractGenericAliasedAbility;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.skills.CAbilityNoTargetSpellBase;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.targeting.AbilityTarget;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.types.definitions.impl.AbilityFields;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.abilities.types.definitions.impl.AbstractCAbilityTypeDefinition;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.timers.CTimer;
+import com.etheller.warsmash.viewer5.handlers.w3x.simulation.trigger.enumtypes.CEffectType;
+
+public class CAbilityReincarnation extends CAbilityNoTargetSpellBase {
```

Comment:
The type `CAbilityNoTargetSpellBase` is an active ability type -- "no target" meaning an instant cast, like an immediate order. If you want an example of something that uses `CAbilitySpell` interface to be "easier to write," but that also is passive, I think that `CAbilityEntangledMine` might be the only one doing that right now (It is the "Entangled gold mine ability"). If you extend `CAbilityNoTargetSpellBase` it is going to inherit the behavior of having a clickable icon, but if you implement the `CAbilitySpell` interface and then reject all order IDs and declare it passive in the callback system, it will not have a clickable icon ingame.

Maybe there should be `CAbilityPassiveSpellBase` as another one of those subclass-able sample types. That could be useful for someone to add that. It could probably be copied from `CAbilityNoTargetSpellBase` but have a different type hierarchy that did not extend `AbstractGenericSingleIconNoSmartActiveAbility` from the older base types that don't have "spell" in the name and don't have the ease-of-use function "populateData" and thus each require their own corresponding AbilityType and AbilityTypeDefinition. Most passive abilities in Warsmash currently extend `AbstractGenericNoIconAbility` but as such do not have access to the `CAbilitySpell` ease-of-use interface.

Reply:
tldr; maybe use `extends CAbilityPassiveSpellBase` and then declare a new class `CAbilityPassiveSpellBase` that is declared with `extends AbstractGenericNoIconAbility implements CAbilitySpell`